migrate/migrations.go

```diff
@@ -18,6 +18,14 @@ import (
 	"github.com/uptrace/bun"
 )
 
+var log = func(format string, a ...interface{}) {
+	fmt.Printf(format, a...)
+}
+
+func SetLogFn(fn func(format string, a ...interface{})) {
+	log = fn
+}
+
```

Comment:
Can you please change this to be an interface with a single method?

```go
type Printer interface {
    Printf(format string, args ...interface{})
}
```

It is easier to extend/change interface-based API in future (e.g. add/remove methods).

Reply:
Yes, exactly that (except replacing `logger` with `printer` in few places). Ideally, it should be a field on some struct, but a global printer is fine too if that is not possible.